tools/slang-test/main.cpp

```diff
@@ -1164,12 +1164,20 @@ TestResult doComputeComparisonTestRunImpl(TestInput& input, const char * langOpt
 	auto referenceProgramOutput = Split(File::ReadAllText(referenceOutput), '\n');
 	if (actualProgramOutput.Count() < referenceProgramOutput.Count())
 		return kTestResult_Fail;
-	for (int i = 0; i < referenceProgramOutput.Count(); i++)
+	for (int i = 0; i < (int)referenceProgramOutput.Count(); i++)
 	{
-		auto reference = StringToFloat(referenceProgramOutput[i]);
-		auto actual = StringToFloat(actualProgramOutput[i]);
-		if (abs(actual - reference) > 1e-7f)
-			return kTestResult_Fail;
+		auto reference = referenceProgramOutput[i];
+		auto actual = actualProgramOutput[i];
+        if (actual != reference)
+        {
+            // try to parse reference as float, and compare again
+            auto val = StringToFloat(reference);
+            auto uval = String((unsigned int)FloatAsInt(val), 16).ToUpper();
+            if (actual != uval)
+			    return kTestResult_Fail;
```

Comment:
Doesn’t this switch from approximate to exact matching for floats? It seems like we’ll need something fuzzy if/when we start testing transcendental functions, etc.

Reply:
This is to allow expected output file written in either binary (uint) form or in floating point form. It is more convenient to author an expected result file in floats instead of hex integers.